src/rgw/rgw_req_context.h

```diff
@@ -0,0 +1,18 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#include "common/async/yield_context.h"
+namespace opentelemetry::trace {
+  class Span;
+}
+
+class DoutPrefixProvider;
+
+// this struct holds information which is created at the frontend
+// and should trickle down through all function calls to the backend
+struct req_context {
+  const DoutPrefixProvider* dpp{nullptr};
+  optional_yield y{null_yield};
```

Comment:
hmm, we've tried hard to make all uses of `null_yield` explicit so we can find and fix them all with grep. defaulting to `null_yield` here makes it possible for a code path to be synchronous without any mention of `null_yield` in its stack

what do you think about removing this default? i'd expect that to disable `req_context`'s default ctor (because `optional_yield` doesn't have a default ctor), so should require callers to pass one into the aggregate initializer like `req_context{.y = null_yield}` or `const req_context rctx{this, s->yield, s->trace.get()}`